src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -77,37 +64,68 @@ public Pivot(PivotIO pivot) {
                       .angularPosition(Radians.of(pivot.getPosition().getRadians()))
                       .angularVelocity(RadiansPerSecond.of(pivot.getVelocity()));
                 },
-                this,
-                "pivot"));
+                this));
 
-    pivotPID.setTolerance(POSITION_TOLERANCE.in(Radians));
-    climbPID.setTolerance(POSITION_TOLERANCE.in(Radians));
+    pid.setTolerance(POSITION_TOLERANCE.in(Radians));
 
     SmartDashboard.putData("pivot quasistatic forward", quasistaticForward());
     SmartDashboard.putData("pivot quasistatic backward", quasistaticBack());
     SmartDashboard.putData("pivot dynamic forward", dynamicForward());
     SmartDashboard.putData("pivot dynamic backward", dynamicBack());
 
-    setDefaultCommand(runPivot(() -> STARTING_ANGLE).repeatedly());
+    setDefaultCommand(run(this::update).repeatedly());
   }
 
   /**
-   * Smoothly angle the pivot to a desired goal using a {@link ProfiledPIDController}.
+   * Smoothly angle the pivot to a desired goal using a {@link ProfiledPIDController} as a proxy to
+   * avoid command composition requirement conflicts.
    *
    * @param goalAngle The position to move the pivot to.
    * @return The command to set the pivot's angle.
    */
   public Command runPivot(Supplier<Rotation2d> goalAngle) {
     return run(() -> {
           double feedback =
-              pivotPID.calculate(pivot.getPosition().getRadians(), goalAngle.get().getRadians());
+              pid.calculate(pivot.getPosition().getRadians(), goalAngle.get().getRadians());
+          double feedforward =
+              ff.calculate( // add pi to measurement to account for alternate angle
+                  pid.getSetpoint().position + Math.PI, pid.getSetpoint().velocity);
+          pivot.setVoltage(feedback + feedforward);
+        })
+        .until(pid::atGoal)
+        .withName("running pivot")
+        .asProxy();
+  }
+
+  /**
+   * Smoothly angle the pivot to its starting position using a {@link ProfiledPIDController}.
+   *
+   * @return The command to set the pivot's angle.
+   */
+  private Command update() {
```

Comment:
update should take in a `Rotation2d`, same as `runPivot`, since you should use it in both places to avoid duplicating code